front/lib/api/assistant/generation.ts

```diff
@@ -699,6 +699,7 @@ export async function* runGeneration(
             configurationId: configuration.sId,
             messageId: agentMessage.sId,
             text: tokenEmitter.getContent() ?? "",
+            runId: await dustRunId,
```

Comment:
Hmmm this is not quite right as we're in a loop here and this will emit only the geneation runId but we likely want to store all intermediary tool uses runs as well

Reply:
Yes, I've checked the call and when the dustRunId is resolved - which is after the first event, so we should never be stuck - but this might  in theory be an issue. I can move the code after the event loop, or use a Promise.race to ensure it's stuck here ?

Reply:
So this loop does

-> call to multi-actions-app ==> I get an action it gets executed
-> call to multi-actions-app ==> I get an action it gets executed
-> call to multi-actions-app ==> starts runing tokens

So here we have 3 run Id for the agentMessage

Reply:
Ok, got it, multi-actions is not in this loop but in `agent.ts` - and yes intermediate runs are not stored so far, looking into it. This one `generation.ts` is not used in multi actions

Reply:
Fixed by yielding the agent_actions event with the associated runId event to conversation, and store the list of runIds that lead to the generation of the message. 
We could also gather all agent actions related in the same list instead of storing them independently, it could make things easier to query ? (but loosing the information that a run is linked to a specific action - just that it participated to the message)

Reply:
@spolu In the last commit 656feac35fbf0bfc7fa63a2c55ff5a6878fa0c48 I also push the runId of the actions up to the conversation through `agent_action_success`. We can see if we want to keep the runIds here, in the action tables, or at both places .. ?

Reply:
Def not both places

The actions runId should live on the action. For now there is one by action.

The main agent runIds should be kept on the agent message. There is multiple already today.